src/views/components.ts

```diff
@@ -0,0 +1,105 @@
+// import * as vscode from "vscode";
+import { workspace, TreeDataProvider, TreeItemCollapsibleState, TreeItem, EventEmitter, Uri, ThemeIcon, CancellationToken, ProviderResult } from "vscode";
+
+class CFCComponentItem extends TreeItem {
+	constructor(
+		public readonly label: string,
+		public readonly fullPath: string
+	) {
+		super(label, TreeItemCollapsibleState.None);
+		this.command = {
+			title: "Open CFML Component",
+			command: "cfmlExplorer.openFile",
+			arguments: [Uri.file(fullPath)],
+		};
+		this.contextValue = "cfmlComponent";
+		this.iconPath = new ThemeIcon("file-code"); // Make sure to use the appropiate icons
+	}
+}
+
+export class CFMLFlatPackageProvider implements TreeDataProvider<CFCComponentItem> {
+	private _onDidChangeTreeData = new EventEmitter<CFCComponentItem | undefined>();
+	readonly onDidChangeTreeData = this._onDidChangeTreeData.event;
+
+	constructor() {
+		workspace.onDidCreateFiles((e) => {
+			if (e.files.some(f => f.fsPath.endsWith(".cfc"))) {
+				this._onDidChangeTreeData.fire(undefined);
+			}
+		});
+	}
+
+	getParent?(element: CFCComponentItem): ProviderResult<CFCComponentItem> {
+		throw new Error("Method not implemented.");
+	}
+
+	resolveTreeItem?(item: TreeItem, element: CFCComponentItem, token: CancellationToken): ProviderResult<TreeItem> {
+		throw new Error("Method not implemented.");
+	}
+
+	// getParent?(element: CFCComponentItem): ProviderResult<CFCComponentItem> {
+	// 	throw new Error("Method not implemented.");
+	// }
+
+	// resolveTreeItem?(item: TreeItem, element: CFCComponentItem, token: CancellationToken): ProviderResult<TreeItem> {
+	// 	throw new Error("Method not implemented.");
+	// }
+
+	setFilter(filter: string) {
```

Comment:
Add // eslint-disable-next-line @typescript-eslint/no-unused-vars or remove unused arguments
```suggestion
	// eslint-disable-next-line @typescript-eslint/no-unused-vars or remove unused arguments 
	setFilter(filter: string) {
```